src/models/forms.js

```diff
@@ -3,13 +3,31 @@
 import neuburgForm from './forms/neuburgForm'
 import mongoose from 'mongoose'
 import cityConfigs from '../cities/cityConfigs'
+import _ from 'lodash'
+import Offer from './Offer'
+
+type AllFormsType = | typeof neuburgForm;
 
 const schemaOptions = {strict: 'throw'}
 
 const Neuburg = mongoose.model(
   cityConfigs.neuburgschrobenhausenwohnraum.cmsName, mongoose.Schema(neuburgForm, schemaOptions)
 )
 
+const addNotIncludedFor = (offer: Offer, form: AllFormsType, names: Array<string>, omit: Array<string>) => {
+  names.forEach((name: string) => {
+    const diff = _.difference(_.get(form, name)['enum'], _.get(offer, `formData.${name}`))
```

Comment:
I really don't like the get and set from lodash. It circumvents typechecking completly.

I'd go with `offer.formData[name]`.

Reply:
I see your point, but you suggestion is not working. The problem is that `name` is a path to a nested object, like 'accomodation.ofRooms'. I haven't found a way to do this in plain javascript, hence I use lodash's functions. Do you know a way to solve this?

Reply:
Hmm... I see... We should at least throw exceptions, if we find, that a target is not of the expected shape... Also `paths` would be more appropriate than `names`:
My suggestion:
```js
const addNotIncludedFor = (offer: Offer, form: AllFormsType, paths: Array<string>, omit: Array<string>) => {
  paths.forEach((path: string) => {
    const arrayConfig = get(form, path)
    if (!arrayConfig || !Array.isArray(arrayConfig.enum) {
      throw Exception(`The supplied form has not the right shape for path '${path}'!`)
    }
    const offerArray = get(offer.formData, path)
    if (!Array.isArray(offerArray)) {
      throw Exception(`The supplied offer has no array at path '${path}'!`)
    }
    const diff = difference(arrayConfig.enum, offerArray)
    pull(diff, ...omit)
    set(offer.formData, `${path}Diff`, diff)
  })
}
```
(Not tested)